src/mmrelay/meshtastic_utils.py

```
@@ -407,22 +444,45 @@ def connect_meshtastic(passed_config=None, force_connect=False):
+                return None
+            if attempts <= retry_limit:
+                wait_time = min(2**attempts, 60)
+                logger.warning(
+                    "Connection attempt %s timed out: %s. Retrying in %s seconds...",
+                    attempts,
+                    e,
+                    wait_time,
+                )
+                time.sleep(wait_time)
+            else:
+                logger.exception(f"Connection failed after {attempts} attempts: {e}")
```

Ruff (0.13.1)

Redundant exception object included in `logging.exception` call

(TRY401)